exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ScanBatch.java

```diff
@@ -237,6 +238,18 @@ private IterOutcome internalNext() {
       logger.trace("currentReader.next return recordCount={}", recordCount);
       Preconditions.checkArgument(recordCount >= 0, "recordCount from RecordReader.next() should not be negative");
       boolean isNewSchema = mutator.isNewSchema();
+      // adds additional record for the case of making scan for obtaining metadata if required
+      if (implicitValues != null) {
+        String projectMetadataColumn = context.getOptions().getOption(ExecConstants.IMPLICIT_PROJECT_METADATA_COLUMN_LABEL).string_val;
+        if (recordCount > 0) {
+          // sets implicit value to false to signalize that some results were returned and there is no need for creating additional record
```

Comment:
signalize --> signal

Maybe add a comment somewhere explaining this "additional record" concept. On the surface, it seems odd. Either a) we are scanning the data and must use the schema from the data to achieve a consistent result, or b) we are gathering stats, which will have its own schema, whatever it is.

I'm struggling to see when we'd want to add some "additional" "record" to a scan. With the same schema as other records? Or, with a different schema? Really could use explanation, please.

Reply:
Thanks, updated the comment and added more details.

Regarding the concept of the additional record, I will try to explain how Metastore collects the data in general cases, it may help to understand the reason for such a decision.

Drill Metastore may collect metadata for every file or row group, so aggregation calls for every column with grouping by `fqn`, `rgi`, `dirX`... columns were added.
This approach works fine for the case of non-empty files and row groups, but when an empty file is present, no data is passed to the aggregation from the Scan, so Metastore was ignoring such files.
To solve this problem, I have added this logic to return a single record for the case when no data was read with the correct values of implicit columns, and this additional implicit column helps to distinguish such records and collect info about rows count, schema, etc.